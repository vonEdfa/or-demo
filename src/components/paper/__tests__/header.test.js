import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import 'jest-styled-components';
import TestRenderer from 'react-test-renderer';

import Header from '../header';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Header', () => {
  it('renders correctly', () => {
    const withCamelCaseSurname = TestRenderer.create(<Header/>);
    expect(withCamelCaseSurname.toJSON()).toMatchSnapshot();
  });
});