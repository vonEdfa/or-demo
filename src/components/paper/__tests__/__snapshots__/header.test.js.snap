// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c4 {
  position: relative;
}

.c4:nth-child(2) {
  margin-top: 3.5em;
}

.c5 {
  padding: 0;
  margin: 0;
  line-height: 1em;
  position: absolute;
  top: -1.2em;
  left: 0;
  text-transform: uppercase;
  font-family: 'Arimo',sans-serif;
  font-weight: normal;
  font-size: 12pt !important;
  color: #a0a5a7;
  z-index: 10;
}

.c6 {
  padding: 0;
  margin: 0;
  font-size: 70pt !important;
  line-height: 0.74em;
  font-family: 'Roboto Slab',serif;
  font-weight: normal;
  font-size: 72pt;
  position: relative;
  left: -0.01em;
}

.c0 {
  width: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  overflow: hidden;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #232526;
  color: #e6e6e6;
}

.c0 p {
  margin: 0;
  padding: 0;
}

.c1 {
  width: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  overflow: hidden;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  margin: 0;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  max-height: 50%;
  text-transform: uppercase;
  font-family: 'Arimo',sans-serif;
  font-weight: bold;
  font-size: 12pt;
}

.c2 p:first-child {
  font-family: 'Roboto Slab',serif;
  font-weight: normal;
  font-size: 14pt;
}

.c2 p:after {
  content: '.';
}

.c2 p {
  padding-bottom: 0.25em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  max-height: 50%;
  font-family: 'Arimo',sans-serif;
  font-weight: normal;
  font-size: 10pt;
  color: #a0a5a7;
}

.c3 p,
.c3 a {
  padding-bottom: 0.5em;
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <p>
        Foo
      </p>
      <p>
        Bar
      </p>
      <p>
        Baz
      </p>
    </div>
    <div
      className="c3"
    >
      <p>
        @tweetMe
      </p>
      <p>
        contact@example.com
      </p>
    </div>
  </div>
  <div
    className="c4"
  >
    <h1
      className="c5"
    >
      Name
    </h1>
    <h1
      className="c6"
    >
      Surname
    </h1>
  </div>
</header>
`;
